src/main.rs

```diff
@@ -1,32 +1,40 @@
+use std::error;
 use std::process::exit;
 
 use async_graphql::{EmptyMutation, EmptySubscription, Schema};
 use axum::routing::post;
 use axum::{Extension, Router};
 
 mod clients;
+mod config;
 mod handlers;
 mod schemas;
 
 use crate::clients::tiled_client::TiledClient;
+use crate::config::GlazedConfig;
 use crate::handlers::graphql::graphql_handler;
 use crate::schemas::TiledQuery;
 
 #[tokio::main]
-async fn main() {
-    let schema = Schema::build(TiledQuery(TiledClient), EmptyMutation, EmptySubscription).finish();
+async fn main() -> Result<(), Box<dyn error::Error>> {
+    let config = GlazedConfig::from_file("config.toml")?;
+
+    let schema = Schema::build(
+        TiledQuery(TiledClient {
+            address: config.tiled_client.address.to_owned(),
+        }),
+        EmptyMutation,
+        EmptySubscription,
+    )
+    .finish();
 
     let app = Router::new()
         .route("/graphql", post(graphql_handler::<TiledClient>))
         .layer(Extension(schema));
 
-    let Ok(listener) = tokio::net::TcpListener::bind("0.0.0.0:3000").await else {
-        eprintln!("Failed to bind TCP Listener");
-        exit(1);
-    };
+    let listener = tokio::net::TcpListener::bind(config.bind_address).await?;
+
+    axum::serve(listener, app).await?;
 
-    let Ok(_) = axum::serve(listener, app).await else {
-        eprintln!("Failed to serve");
-        exit(1);
-    };
+    exit(0);
```

Comment:
[exit()](https://doc.rust-lang.org/std/process/fn.exit.html) is a bit brutal and given we're running inside tokio's event loop here, it's probably good to let it clean up.

```suggestion
    Ok(())
```